sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/state/RocksDB.scala

```diff
@@ -1337,6 +1389,13 @@ class RocksDB(
     rocksDBMetricsOpt
   }
 
+  /**
+   * Refresh the recorded metrics with the latest metrics.
+   */
+  def refreshRecordedMetrics(): Unit = {
```

Comment:
is this for test only? if so make it private[state] and name `refreshRecordedMetricsForTest`